File: 프로그래머스/1/42748. K번째수/K번째수.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    vector<int> sub;
    int i, j, k;
    for(int m=0 ; m<commands.size() ; m++){
        i=commands[m][0]-1;
        j=commands[m][1]-1;
        k=commands[m][2]-1;
        sub={}; //초기화
        
        for(int n=i ; n<=j ; n++){
            sub.push_back(array[n]);
        }
        sort(sub.begin(), sub.end());
        answer.push_back(sub[k]);
    }
    
    return answer;
}

File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = new int[commands.length];
        for(int i =0 ; i<commands.length ; i++){
            // array의 2번째부터 5번째까지 자르면 [5, 2, 6, 3]입니다.
            int[] tempArray = Arrays.copyOfRange(array, commands[i][0]-1, commands[i][1]);

            // 1에서 나온 배열을 정렬하면 [2, 3, 5, 6]입니다.
            Arrays.sort(tempArray);

            // 2에서 나온 배열의 3번째 숫자는 5입니다.
            answer[i] = tempArray[commands[i][2] - 1];

        }
        return answer;
    }
}